ydb/services/ydb/ydb_login_ut.cpp

```diff
@@ -0,0 +1,158 @@
+            UNIT_ASSERT_STRINGS_EQUAL_C(response->Record.GetPathDescription().GetSelf().GetName(), "Root", response->Record.ShortDebugString());
+        } else {
+            UNIT_ASSERT_C(response->Record.GetStatus() != NMsgBusProxy::MSTATUS_OK, response->Record.ShortDebugString());
+            UNIT_ASSERT_C(response->Record.GetStatusCode() != NKikimrIssues::TStatusIds::SUCCESS, response->Record.ShortDebugString());
+            UNIT_ASSERT_STRINGS_EQUAL_C(response->Record.GetErrorReason(), expectedErrorReason, response->Record.ShortDebugString());
+        }
+    }
+
+private:
+    NKikimrConfig::TAppConfig InitAuthSettings(bool isLoginAuthenticationEnabled = true) {
+        NKikimrConfig::TAppConfig appConfig;
+        auto authConfig = appConfig.MutableAuthConfig();
+
+        authConfig->SetUseBlackBox(false);
+        authConfig->SetUseLoginProvider(true);
+        authConfig->SetEnableLoginAuthentication(isLoginAuthenticationEnabled);
+        appConfig.MutableDomainsConfig()->MutableSecurityConfig()->SetEnforceUserTokenRequirement(true);
+        appConfig.MutableFeatureFlags()->SetAllowYdbRequestsWithoutDatabase(false);
+        appConfig.MutableFeatureFlags()->SetCheckDatabaseAccessPermission(true);
+
+        return appConfig;
+    }
+
+    TDriverConfig GetDriverConfig(ui16 grpcPort) {
+        TDriverConfig config;
+        config.SetEndpoint("localhost:" + ToString(grpcPort));
+        return config;
+    }
+
+private:
+    TKikimrWithGrpcAndRootSchemaWithAuth Server;
+    NYdb::TDriver Connection;
+    NConsoleClient::TDummyClient Client;
+};
+
+} // namespace
+
+
+Y_UNIT_TEST_SUITE(TGRpcAuthentication) {
+    const TString User = "user";
+    const TString Password = "UserPassword";
+
+    Y_UNIT_TEST(ValidCredentials) {
+        TLoginClientConnection loginConnection;
+        loginConnection.CreateUser(User, Password);
+        loginConnection.ModifyACL(true, User, NACLib::EAccessRights::ConnectDatabase | NACLib::EAccessRights::DescribeSchema);
+        auto factory = CreateLoginCredentialsProviderFactory({.User = User, .Password = Password});
+        auto loginProvider = factory->CreateProvider(loginConnection.GetCoreFacility());
+        TString token;
+        UNIT_ASSERT_NO_EXCEPTION(token = loginProvider->GetAuthInfo());
+        UNIT_ASSERT(!token.empty());
+
+        loginConnection.Ls(token, "");
+
+        loginConnection.Stop();
+    }
+
+    Y_UNIT_TEST(InvalidPassword) {
+        TLoginClientConnection loginConnection;
+        loginConnection.CreateUser(User, Password);
+
+        auto factory = CreateLoginCredentialsProviderFactory({.User = User, .Password = "WrongPassword"});
+        auto loginProvider = factory->CreateProvider(loginConnection.GetCoreFacility());
+        UNIT_ASSERT_EXCEPTION_CONTAINS(loginProvider->GetAuthInfo(), yexception, "Invalid password");
+
+        loginConnection.Stop();
+    }
+
+    Y_UNIT_TEST(DisableBuiltinAuthMechanism) {
```

Comment:
```suggestion
    Y_UNIT_TEST(DisableLoginAuthentication) {
```
Builtin auth mechanist it is about tokens with `@builtin` domain

Reply:
Done